homeassistant/components/deluge/__init__.py

```diff
@@ -49,20 +50,16 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             ) from ex
         _LOGGER.error("Unknown error connecting to Deluge: %s", ex)
 
-    coordinator = DelugeDataUpdateCoordinator(hass, api, entry)
-    await coordinator.async_config_entry_first_refresh()
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
+    entry.runtime_data = DelugeDataUpdateCoordinator(hass, api, entry)
+    await entry.runtime_data.async_config_entry_first_refresh()
```

Comment:
```suggestion
    coordinator = DelugeDataUpdateCoordinator(hass, api, entry)
    await coordinator.async_config_entry_first_refresh()
    entry.runtime_data = coordinator
```

See https://github.com/home-assistant/core/pull/119551#discussion_r1637422489